computation.py

```diff
@@ -0,0 +1,142 @@
+            "month": min_temp_month_month,
+            "year": min_temp_month_year,
+        }
+
+        return max_monthly, min_monthly, mean_humidity_month
+
+    def strip_date(self, date):
+        """Strips the date into correct format and returns the day, month, and year."""
+        if date is not None:
+            date_refined = datetime.datetime.strptime(date, "%Y-%m-%d")
+            date_day = date_refined.strftime("%d")
+            date_month = date_refined.strftime("%B")
+            date_year = date_refined.strftime("%Y")
+            return date_day, date_month, date_year
+        else:
+            return None, None, None
+
+    def compute_yearly_data(self, yearly_data):
+        """checks whether any data is None , if so , removes it .
+        Computes the yearly Maximum Temperature , Minimum Temperature , Maxiimum Humidity and their Corresponding Dates
+        Returns 3 dictionaries of Maxtemp , Min temp and Max humidity"""
+
+        max_temp = float("-inf")
+        min_temp = float("inf")
+        max_humidity = float("-inf")
+        max_temp_date = None
+        min_temp_date = None
+        max_humidity_date = None
+
+        # Iterate through each month's data in the yearly_data
+        for monthly_data in yearly_data:
+            for row in monthly_data:
+                if row.max_temp is not None and isinstance(row.max_temp, (int, float)):
+                    if row.max_temp > max_temp:
+                        max_temp = row.max_temp
+                        max_temp_date = row.pkt
+
+                if row.min_temp is not None and isinstance(row.min_temp, (int, float)):
+                    if row.min_temp < min_temp:
+                        min_temp = row.min_temp
+                        min_temp_date = row.pkt
+
+                if row.max_humidity is not None and isinstance(
+                    row.max_humidity, (int, float)
+                ):
+                    if row.max_humidity > max_humidity:
+                        max_humidity = row.max_humidity
+                        max_humidity_date = row.pkt
+
+        max_temp_day, max_temp_month, max_temp_year = self.strip_date(max_temp_date)
+        min_temp_day, min_temp_month, min_temp_year = self.strip_date(min_temp_date)
+        max_humidity_day, max_humidity_month, max_humidity_year = self.strip_date(
+            max_humidity_date
+        )
+
+        max_temp_overall = {
+            "temperature": max_temp,
+            "day": max_temp_day,
+            "month": max_temp_month,
+            "year": max_temp_year,
+        }
+
+        min_temp_overall = {
+            "temperature": min_temp,
+            "day": min_temp_day,
+            "month": min_temp_month,
+            "year": min_temp_year,
+        }
+
+        max_humidity_overall = {
+            "humidity": max_humidity,
+            "day": max_humidity_day,
+            "month": max_humidity_month,
+            "year": max_humidity_year,
+        }
```

Comment:
Can we not move these fields inside WeatherDataRow and assign WeatherDataRow object to these 3 variables?